pkg/profile/artifact.go

```diff
@@ -6,28 +6,51 @@ import (
 	"strings"
 	"time"
 
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
 	kustomizev1 "github.com/fluxcd/kustomize-controller/api/v1beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
-	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
-	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
 )
 
+// location defines where the helm chart is located.
+// remote -> in a helm repository
+// local -> in the profile repository
+type location int
+
+var local location = 0
+var remote location = 1
+
 // CreateArtifacts creates and inserts objects to the cluster to deploy the
 // profile as a HelmRelease.
 func (p *Profile) CreateArtifacts(ctx context.Context) error {
-	if err := p.createGitRepository(ctx); err != nil {
-		return fmt.Errorf("failed to create GitRepository resource: %w", err)
-	}
-
+	// TODO don't depend on 0.
```

Comment:
We could expand this to `Artifacts[0]` ?

Reply:
I'm not sure I follow, sorry. The `[0]` will go away. Jake has a branch somewhere where he's fixing the reliance on it. We have to enumerate the artifacts anyways. So this is just a note to not use this at all.

Reply:
https://github.com/weaveworks/profiles/pull/61